### PR TITLE
Build linux arm64 in ci

### PR DESCRIPTION
**Pull request by cap10morgan**
Please answer the following questions and leave the below in as part of your PR.

- [X] I have read the [developer documentation](https://github.com/clj-kondo/clj-kondo/blob/master/doc/dev.md).

- [X] This PR corresponds to an [issue with a clear problem statement](https://github.com/clj-kondo/clj-kondo/blob/master/doc/dev.md#start-with-an-issue-before-writing-code). #1238 

- [ ] This PR contains a [test](https://github.com/clj-kondo/clj-kondo/blob/master/doc/dev.md#tests) to prevent against future regressions. N/A for this one, I think?

- [X] I have updated the [CHANGELOG.md](https://github.com/clj-kondo/clj-kondo/blob/master/CHANGELOG.md) file with a description of the addressed issue.

### CODE REVIEW

**Comment by borkdude** [comment]
Makes sense! I'm surprised we still didn't have an aarch build for clj-kondo. I think it's best to go with the static-except-glibc approach by default, like we do with babashka in the aarch64-static one.

**Comment by borkdude** [comment]
The compile script doesn't do that yet for aarch64: https://github.com/clj-kondo/clj-kondo/blob/e556f9c5f53620cc9acb2030257c1461556dab84/script/compile#L25
Compare to babashka's compile script.

**Comment by cap10morgan** [comment]
> The compile script doesn't do that yet for aarch64:
> 
> https://github.com/clj-kondo/clj-kondo/blob/e556f9c5f53620cc9acb2030257c1461556dab84/script/compile#L25

It doesn't? I'm setting `CLJ_KONDO_STATIC` to `"true"` and I see this in the binary build output in CI: https://app.circleci.com/pipelines/github/cap10morgan/clj-kondo/3/workflows/43097adf-35a4-4fc3-9213-9e66f51a2aa1/jobs/10?invite=true#step-107-191

File: .circleci/script/tools.deps [hunk 1]
```diff
@@ -24,7 +24,8 @@ rm -rf /tmp/proj
 mkdir -p /tmp/proj
 cd /tmp/proj
 
-github_user=${CIRCLE_PR_USERNAME:-borkdude}
+github_user=${CIRCLE_PR_USERNAME} # for PRs
+github_user=${CIRCLE_PROJECT_USERNAME:-borkdude} # for non-PR forks
```
[the remark this answers is not in the document]
Comment (borkdude) [reply .circleci/script/tools.deps:28]: Won't the second line overwrite the first line?

**Comment by borkdude** [comment]
Yes, but static binaries linked against the system's glibc aren't recommended, unless you use musl. 

See https://github.com/babashka/babashka/blob/9241c8123878021390442ba483239f57553786fe/script/compile#L69-L72

File: .circleci/script/tools.deps (hunk 1 above)
[the remark this answers is not in the document]
Comment (cap10morgan) [reply .circleci/script/tools.deps]: Yep, good catch. Hopefully fixed in [b9e4572](https://github.com/clj-kondo/clj-kondo/pull/1837/commits/b9e45727c69c2be225f0db077b08e1d583066669).

**Comment by cap10morgan** [comment]
> Yes, but static binaries linked against the system's glibc aren't recommended, unless you use musl.
> 
> See https://github.com/babashka/babashka/blob/9241c8123878021390442ba483239f57553786fe/script/compile#L69-L72

OK added that in [93c3d3a](https://github.com/clj-kondo/clj-kondo/pull/1837/commits/93c3d3a6d4e79170b150fe53dd3ec31d930ee550).

**Comment by borkdude** [comment]
Terrific, thanks!

**Pull request merged**